jina/peapods/pod.py

```diff
@@ -556,6 +556,25 @@ def start(self) -> 'GatewayPod':
         return self
 
 
+class AsyncGatewayPod(BasePod, AsyncExitStack):
```

Comment:
Will this play nice with inheriting both `ỀxitStack` and `AsyncExitStack` (I think is a good point I may try to have a mixin in the refactor for `ExitStack` so that it is more separated from `BasePod`.

Reply:
```

class DualContext(ExitStack, AsyncExitStack):
    def __init__(self) -> None:
        super(ExitStack, self).__init__()
        super(AsyncExitStack, self).__init__()

    def start(self):
        raise NotImplementedError

    def close(self):
        raise NotImplementedError

    async def astart(self):
        raise NotImplementedError

    async def aclose(self):
        raise NotImplementedError

    def __enter__(self):
        return self.start()

    def __exit__(self, exc_type, exc_val, exc_tb):
        self.close()
        super(ExitStack, self).__exit__(exc_type, exc_val, exc_tb)

    async def __aenter__(self):
        return await self.astart()

    async def __aexit__(self, exc_type, exc_val, exc_tb):
        await self.aclose()
        await super(AsyncExitStack, self).__aexit__(exc_type, exc_val, exc_tb)


class SyncContext(DualContext):
    def start(self):
        print('Sync')

    def close(self):
        print('Sync close')


class AsyncContext(DualContext):
    async def astart(self):
        print('Async')

    async def aclose(self):
        print('Async close')


with SyncContext():
    pass

async def blah():
    async with AsyncContext():
        pass

asyncio.run(blah())
```

Reply:
This works, maybe we can use something similar